reports/api.py

```diff
@@ -178,6 +186,13 @@ class ChildViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
             ),
         )
         .prefetch_related("guardians")
+        .prefetch_related("guardians__user")
         .order_by("name")
     )
     serializer_class = ChildSerializer
+
+    def get_queryset(self):
+        filters = {}
+        if is_obsolete := self.request.query_params.get("is_obsolete"):
+            filters["guardians__user__is_obsolete"] = bool(strtobool(is_obsolete))
+        return super().get_queryset().filter(**filters)
```

Comment:
This will throw a ValueError if is_obsolete query parameter is not valid, e.g. 'maybe':
```python
def strtobool (val):
    """Convert a string representation of truth to true (1) or false (0).

    True values are 'y', 'yes', 't', 'true', 'on', and '1'; false values
    are 'n', 'no', 'f', 'false', 'off', and '0'.  Raises ValueError if
    'val' is anything else.
    """
    val = val.lower()
    if val in ('y', 'yes', 't', 'true', 'on', '1'):
        return 1
    elif val in ('n', 'no', 'f', 'false', 'off', '0'):
        return 0
    else:
        raise ValueError("invalid truth value %r" % (val,))
```
which will result in internal server error probably? Is that acceptable or should the invalid values rather fall back to not taking the parameter into account at all? How do you see this?

Reply:
BTW [distutils is deprecated](https://peps.python.org/pep-0632/) and was removed in Python 3.12.

Reply:
I copied this function and used try-except